syntax = "proto2";

package pb;

message E2Config{
	required uint32 N = 1;
	optional double A = 2;
	optional double delta = 3;
	optional double f = 4;
}

message E2State{
	repeated group Particles = 1{
		required double ksi = 2;
		required double v = 3;
	}
}

message E2PetscSolverConfig{
	optional double tolerance = 1;
	optional double init_step = 2;
}